src/Raven.Server/Utils/TypeConverter.cs

```diff
@@ -142,6 +144,47 @@ private static object ToBlittableSupportedType(object root, object value, bool f
             {
                 NestingLevelTooDeep(root);
             }
+             if (value is JsValue js)
+            {
+                if (js.IsNull() || js.IsUndefined())
+                    return null;
+                if (js.IsString())
+                    return js.AsString();
+                if (js.IsBoolean())
+                    return js.AsBoolean().ToString(); // avoid boxing the boolean
```

Comment:
This should be just a bool. Without .ToString()

Reply:
This is done intentionally. When using this for indexing, we index `true` as the string `True`
Because this method returns an `object`, if we return a `Boolean` here it will be boxed. So we cut the middleman and avoid the boxing allocation. The end result is that the string `True` is indexed.
Note that this is an optimization only, and intended only for indexing.

Reply:
Ok, although I do not like the idea of having a different flow inside LuceneDocumentConverterBase for bools from JSIndex and any other type of index.

Reply:
I'm fine with making this change if this code is now reused, mind.

Reply:
Ok. Then let's remove it.